test/celeritas/track/TrackSort.test.cc

```diff
@@ -0,0 +1,179 @@
+
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2022-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/track/TrackSort.test.cc
+//---------------------------------------------------------------------------//
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+#include "corecel/data/Collection.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/ext/GeantPhysicsOptions.hh"
+#include "celeritas/global/CoreTrackData.hh"
+#include "celeritas/global/Stepper.hh"
+#include "celeritas/phys/PDGNumber.hh"
+#include "celeritas/phys/ParticleParams.hh"
+#include "celeritas/phys/Primary.hh"
+#include "celeritas/track/TrackInitParams.hh"
+#include "celeritas/track/detail/TrackSortUtils.hh"
+
+#include "../TestEm3Base.hh"
+#include "../global/StepperTestBase.hh"
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace test
+{
+
+using celeritas::units::MevEnergy;
+
+#define TestTrackPartitionEm3Stepper \
+    TEST_IF_CELERITAS_GEANT(TestTrackPartitionEm3Stepper)
+class TestTrackPartitionEm3Stepper : public TestEm3Base, public StepperTestBase
```

Comment:
This is  `TestEm3NoMsc` from `Stepper.test.cc` except that we need to override `build_init`.  We could move `TestEm3StepperTestBase` or `TestEm3NoMsc` out of the cc file and inherit from it?

Reply:
Yeah, why don't you just make a `TestEm3NoMscStepper.hh` (or whatever) with the common class, and override the `build_init` in this test?